tests/rules/test_new_lines.py

```diff
@@ -58,3 +61,27 @@ def test_dos_type(self):
         self.check('\r\n', conf)
         self.check('---\ntext\n', conf, problem=(1, 4))
         self.check('---\r\ntext\r\n', conf)
+
+    def test_platform_type(self):
+        conf = ('new-line-at-end-of-file: disable\n'
+                'new-lines: {type: platform}\n')
+
+        self.check('', conf)
+
+        # mock the Linux new-line-character
+        with mock.patch("yamllint.rules.new_lines.linesep", '\n'):
+            self.check('\n', conf)
+            self.check('\r\n', conf, problem=(1, 1))
+            self.check('---\ntext\n', conf)
+            self.check('---\r\ntext\r\n', conf, problem=(1, 4))
+            self.check('---\r\ntext\n', conf, problem=(1, 4))
+            self.check('---\ntext\r\n', conf, problem=(2, 4))
+
+        # mock the Windows new-line-character
+        with mock.patch("yamllint.rules.new_lines.linesep", '\r\n'):
+            self.check('\r\n', conf)
+            self.check('\n', conf, problem=(1, 1))
+            self.check('---\r\ntext\r\n', conf)
+            self.check('---\ntext\n', conf, problem=(1, 4))
+            self.check('---\ntext\r\n', conf, problem=(1, 4))
+            self.check('---\r\ntext\n', conf, problem=(2, 4))
```

Comment:
both test return no error when only the end of file doesn't match.

Is that by design? (presumably because the end-of-file rule is handled sepertely?)

Reply:
This is a good question, and I'm not sure I remember perfectly. But I believe *by design*, the `new-lines` rule was made to check only the first line of a file.

In my opinion, this example `---\r\ntext\n` *has* a bug, and either `new-line-at-end-of-file` or `new-lines` should report a problem. But maybe this isn't in the scope of this PR, since this "bug" already exists? What do you think?

Reply:
I also just tested `---\ntext\r\nfoo\n` which also raises no error. And I agree that this should defnetly be caught. So the problem really seems to ba that only the first line is checked.

But as the problem does not come from the actuall rule file (`yamllint/rules/new_lines.py`) but from the calling system, I would consider it out of scope of this PR too.

I will open an issue about this and coment out the tests (with a TODO note), if thats OK with you?

Reply:
I handeld it like this now:

```python
            # FIXME: the following tests currently don't work
            # because only the first line is checked for line-endings
            # see: #
            # ---
            # self.check('---\r\ntext\nfoo\r\n', conf, problem=(2, 4))
            # self.check('---\r\ntext\n', conf, problem=(2, 4))
```

I will add the link to the issue once its opend.

That OK with you?

Reply:
Good idea :+1: